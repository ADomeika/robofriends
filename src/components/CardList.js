import React from 'react';

import Card from './Card';

const CardList = ({ robots }) => {
  return (
    <div>
      {robots.map((robot) => (
        <Card key={robot.id} item={robot} />
      ))}
    </div>
  );
};
export default CardList;
