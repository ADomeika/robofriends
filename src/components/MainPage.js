import React, { Component } from 'react';

import './MainPage.css';
import CardList from './CardList';
import Header from './Header';
import Scroll from './Scroll';
import SearchBox from './SearchBox';
import ErrorBoundry from './ErrorBoundry';

class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    const { robots, searchField } = this.props;
    return robots.filter((robot) =>
      robot.name.toLowerCase().includes(searchField.toLowerCase())
    );
  };

  render() {
    const { onSearchChange, isPending } = this.props;

    return isPending ? (
      <h1 className='tc'>Loading...</h1>
    ) : (
      <div className='tc'>
        <Header />
        <SearchBox searchChange={onSearchChange} />
        <Scroll>
          <ErrorBoundry>
            <CardList robots={this.filterRobots()} />
          </ErrorBoundry>
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
