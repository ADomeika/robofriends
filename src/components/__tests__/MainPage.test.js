import React from 'react';
import { shallow } from 'enzyme';
import MainPage from '../MainPage';
import CardList from '../CardList';

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

describe('<MainPage />', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should filter robots correctly', () => {
    const mockProps2 = {
      onRequestRobots: jest.fn(),
      robots: [
        {
          id: 3,
          name: 'john',
          email: 'john@gmail.com',
        },
      ],
      searchField: 'a',
      isPending: false,
    };
    const wrapper2 = shallow(<MainPage {...mockProps2} />);

    expect(wrapper2.instance().filterRobots()).toEqual([]);
    expect(wrapper2.instance().filterRobots()).toEqual([]);
  });

  it('should filter robots correctly 2', () => {
    const mockProps3 = {
      onRequestRobots: jest.fn(),
      robots: [
        {
          id: 3,
          name: 'john',
          email: 'john@gmail.com',
        },
      ],
      searchField: 'john',
      isPending: false,
    };
    const wrapper3 = shallow(<MainPage {...mockProps3} />);

    expect(wrapper3.instance().filterRobots()).toEqual([
      {
        id: 3,
        name: 'john',
        email: 'john@gmail.com',
      },
    ]);
  });

  it('should not render <CardList /> if isPending is true', () => {
    const mockProps4 = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: 'john',
      isPending: true,
    };
    const wrapper4 = shallow(<MainPage {...mockProps4} />);
    expect(wrapper4.find(<CardList />)).toHaveLength(0);
  });
});
