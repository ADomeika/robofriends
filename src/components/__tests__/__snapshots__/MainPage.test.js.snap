// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> should match snapshot 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
